wolfhsm/wh_server.h

```diff
@@ -39,6 +41,10 @@ typedef struct whServerContext_t {
 typedef struct whServerConfig_t {
     whCommServerConfig* comm_config;
     whNvmConfig* nvm_config;
+#if defined WOLF_CRYPTO_CB /* TODO: should we be relying on wolfSSL defines? */
```

Comment:
devId should ALWAYS be defined and set to invalid_dev_id when no cryptocb is available.  dev_id=0 means server is not initialized yet.

For now, let's reuse the wolfCrypt defines to ensure nothing is out of sync.

Reply:
Keep in mind this is in the config struct - devId is always defined in the server context, since we will always need to pass it into the internally invoked wolfCrypt functions. Here, if we don't have support for crypto callbacks (e.g. `WOLF_CRYPTO_CB` isn't defined) then it doesn't exist in the config struct, and is always stored as `INVALID_DEVID` when passed to wolfCrypt APIs internally (which are then ignored since we haven't compiled in support for CCBs). 

I agree with @jpbland1 that we don't want a secret meaning to devId==0, as that is a perfectly valid devId. Instead I added an additional flag for devId initialization.